app/models/parameter.rb

```diff
@@ -67,14 +68,19 @@ def hash_for_include_source(source, source_name = nil)
     options = {
       :value => value, :source => source, :key_type => key_type,
       :safe_value => safe_value, :parameter_type => parameter_type,
-      :hidden_value? => hidden_value?
+      :hidden_value? => hidden_value?,
+      :cloned_value => cloned_value
     }
     options[:source_name] = source_name if source_name
     options
   end
 
   private
 
+  def set_cloned_value
+    self.cloned_value = Parameter.format_value_before_type_cast(value, key_type)
```

Comment:
Perhaps `string_value` or `searchable_value` would be clearer as the name for this attribute?

Reply:
Renamed it to `searchable_value`.